### DISCUSSION TITLE
Is it possible to propagate CRD with cluster scope ?

### DISCUSSION THREAD

**Discussion by NickYadance** [karmada-io/karmada#3125] [Q&A] [answered]
The [Propagate a CRD application with Karmada](https://karmada.io/zh/docs/tutorials/crd-application) seems only to work with namespaced CRD. The CRD application in cluster scope cannot be propagated.

Karmada version: latest
K8s:
```
Client Version: version.Info{Major:"1", Minor:"26", GitVersion:"v1.26.1", GitCommit:"8f94681cd294aa8cfd3407b8191f6c70214973a4", GitTreeState:"clean", BuildDate:"2023-01-18T15:51:24Z", GoVersion:"go1.19.5", Compiler:"gc", Platform:"darwin/arm64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.4", GitCommit:"e6c093d87ea4cbb530a7b2ae91e54c0842d8308a", GitTreeState:"clean", BuildDate:"2022-03-06T21:39:59Z", GoVersion:"go1.17.7", Compiler:"gc", Platform:"linux/arm64"}
```
Reproduce:
1. Start local Karmada cluster.
2. Create a unnamespaced Guestbook.
```
# namespaced=false
kubebuilder create api --group webapp --version v1 --namespaced=false --kind Guestbook
```
3. Install CRD in karmada and propagate CRD application following the steps in [Propagate a CRD application with Karmada](https://karmada.io/zh/docs/tutorials/crd-application).
4. Get CRD in members
```bash
$ kubectl --kubeconfig=$HOME/.kube/members.config --context=member1 get guestbook                                                                   
No resources found
```
Expected result(work for namespaced CRD):
```bash
$ kubectl --kubeconfig=$HOME/.kube/members.config --context=member1 get guestbook                                                                   
NAME               AGE
guestbook-sample   2s
```
Karmada controller logs
```
I0207 13:49:21.965952       1 detector.go:748] Create PropagationPolicy(policy.karmada.io/v1alpha1, kind=PropagationPolicy, default/example-policy)
I0207 13:49:21.966107       1 detector.go:796] PropagationPolicy(default/example-policy) has been added.
I0207 13:49:21.974679       1 detector.go:978] Matched 0 resources by policy(default/example-policy)

```

**Comment by RainbowMango** [comment]
> To make things clear, i take Configmap for example. The Configmap can be treated as the kind of CRD without replica, and karmada can propagate it in my test. So do i miss anything or is it possible to achieve this?

I don't get it, why are you want to customize your CRD with the `resource interpreter`?

**Reply by NickYadance** [reply]
sry for some misleading question, re-edited already.

**Comment by NickYadance** [comment, marked as answer]
After searching a while in the docs i finally found out that for CRD in cluster scope,  the `ClusterPropagationPolicy` should be applied instead of `PropagationPolicy` , refered to this: [faq](https://karmada.io/docs/faq/)
>What is the difference between PropagationPolicy and ClusterPropagationPolicy?

> The PropagationPolicy is a namespace-scoped resource type which means the objects with this type must reside in a namespace. And the ClusterPropagationPolicy is the cluster-scoped resource type which means the objects with this type don't have a namespace.
Both of them are used to hold the propagation declaration, but they have different capacities:

PropagationPolicy: can only represent the propagation policy for the resources in the same namespace.
ClusterPropagationPolicy: can represent the propagation policy for all resources including namespace-scoped and cluster-scoped resources.

**Reply by RainbowMango** [reply]
Yes, exactly.
